File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WinCtrlData.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WinCtrlData
*/
public class t_WinCtrlData : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_kind;
	public uint	m_propertyNum;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WinCtrlData() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_kind = 0;
		m_propertyNum = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_kind = cVariable.getU32();
			m_propertyNum = cVariable.getU32();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_kind);
		cVariable.put(m_propertyNum);
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WinCtrlData.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_WinCtrlData
#/
class t_WinCtrlData:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_kind = 0	#U32
	m_propertyNum = 0	#U32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_kind = 0
		self.m_propertyNum = 0

	def read(self,cVariable):
		self.m_kind = cVariable.getU32()
		self.m_propertyNum = cVariable.getU32()
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_kind)
		cVariable.putU32(self.m_propertyNum)
		return True

